### PR TITLE
BROWSER-108: branding system for Windows installer

### PR DESCRIPTION
**Pull request by alver-cliqz**
Idea taken from Omaha project. Inject brand label into installer and later, on install stage, create a distribution file in profile folder with it. More information about how this works could be found in wiki

### CODE REVIEW

**Comment by chrmod** [comment]
Is CliqzHelpers.dll compiled along with the rest of the build or we used compiled version stored in the repo?

Will this work with 64 bit build?

**Comment by alver-cliqz** [comment]
All NSIS plugins used as pre-compiled in Firefox build process. You can find source code for each plugin, which installer use, but Firefox use pre-compiled one. I added our plugin in similar way.

About 64bit - i didn't find any special for 64-bit build for Windows. I think this is could be separate tasks, additional to building 64-bit FF version itself.

**Comment by maxim-cliqz** [comment]
Is it possible to add a readme file for ApplyTag.exe, explaining its origin and documenting its usage?

**Comment by alver-cliqz** [comment]
Sure, will do

File: mozilla-release/other-licenses/nsis/Contrib/CliqzHelper/dllmain.cpp [hunk 1]
```diff
@@ -0,0 +1,76 @@
+#include <windows.h>
+#include <shlobj.h>
+#include <stdio.h>
```
Comment (maxim-cliqz) [inline mozilla-release/other-licenses/nsis/Contrib/CliqzHelper/dllmain.cpp]: I'd sort headers by name and also separate standard library headers from system ones,

File: mozilla-release/other-licenses/nsis/Contrib/CliqzHelper/dllmain.cpp [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+// Read key value from parameters, try to find this key in
+// tagged area of parent process, if exist - check does
+// distribution file exist. If not - save information about
+// brand into distribution file (in CSIDL_APPDATA\\Cliqz folder)
+extern "C" void __declspec(dllexport) setBrand(HWND hwndParent,
+                                               int string_size,
+                                               TCHAR *variables,
+                                               stack_t **stacktop) {
+#ifdef _DEBUG
+  MessageBox(hwndParent, L"1", L"2", MB_OK);
```
Comment (maxim-cliqz) [inline mozilla-release/other-licenses/nsis/Contrib/CliqzHelper/dllmain.cpp]: Is this still needed? It introduces extra argument `hwndParent`, not used by the main part of code.

File: mozilla-release/other-licenses/nsis/Contrib/CliqzHelper/dllmain.cpp [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+    if (!value.empty()) {
+      TCHAR szPath[MAX_PATH];
+      if (SUCCEEDED(SHGetFolderPath(NULL, CSIDL_APPDATA | CSIDL_FLAG_CREATE, NULL, 0, szPath))) {
+        wcscat_s(szPath, L"\\CLIQZ");
+        SHCreateDirectory(0, szPath);
+        wcscat_s(szPath, L"\\distribution");
+        DWORD dwAttrib = GetFileAttributes(szPath);
+        BOOL file_exist = (dwAttrib != INVALID_FILE_ATTRIBUTES) && !(dwAttrib & FILE_ATTRIBUTE_DIRECTORY);
+        if (!file_exist) {
+          HANDLE hFile = CreateFile(szPath, GENERIC_WRITE, 0, NULL, CREATE_NEW, FILE_ATTRIBUTE_NORMAL, NULL);
```
Comment (maxim-cliqz) [inline mozilla-release/other-licenses/nsis/Contrib/CliqzHelper/dllmain.cpp]: Seems more than 80 chars

**Pull request merged**